source/common/buffer/buffer_impl.cc

```diff
@@ -43,8 +43,6 @@ void OwnedImpl::addDrainTracker(std::function<void()> drain_tracker) {
 
 void OwnedImpl::bindAccount(BufferMemoryAccountSharedPtr account) {
   ASSERT(slices_.empty());
-  // We don't yet have an account bound.
-  ASSERT(!account_);
```

Comment:
It seems odd that we need to remove this ASSERT, can you provide more context?

Reply:
I see...

And recreate does seem to create a new account on newStream so it's not even the same account.  Ok, good to know.  Thanks for the explanation.